python/lsst/cp/pipe/cpFilterScan.py

```diff
@@ -0,0 +1,377 @@
+        super().__init__(config=config)
+
+        if config and not config.useElectrometer:
+            self.inputs.discard("inputElectrometerHandles")
+
+
+class CpFilterScanTaskConfig(pipeBase.PipelineTaskConfig,
+                             pipelineConnections=CpFilterScanConnections):
+    maskNameList = pexConfig.ListField(
+        dtype=str,
+        doc="Mask list to exclude from statistics calculations.",
+        default=["DETECTED", "BAD", "CR", "NO_DATA", "INTRP"]
+    )
+    useElectrometer = pexConfig.Field(
+        dtype=bool,
+        doc="Use electrometer data?",
+        default=False,
+    )
+    referenceFilter = pexConfig.Field(
+        dtype=str,
+        doc="Filter to use as baseline reference.",
+        default="empty~empty",
+    )
+    combine = pexConfig.ConfigurableField(
+        target=CalibCombineTask,
+        doc="Combine task to use for header merging.",
+    )
+    efdClientInstance = pexConfig.Field(
+        dtype=str,
+        doc="EFD instance to use for monochromator results.",
+        default="usdf_efd",
+    )
+
+
+class CpFilterScanTask(pipeBase.PipelineTask):
+    """Create filter scan from appropriate data.
+    """
+
+    ConfigClass = CpFilterScanTaskConfig
+    _DefaultName = "cpFilterScan"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.makeSubtask("combine")
+
+    def run(self, inputExpHandles, inputElectrometerHandles=None):
+        """Construct filter scan from the header and visit info of processed
+        exposures.
+
+        Parameters
+        ----------
+        inputExpHandles : `list` [`lsst.daf.butler.DeferredDatasetHandle`]
+            Input list of exposure handles to combine.
+        inputElectrometerHandles : `list` [`lsst.daf.butler.DeferredDatasetHandle`], optional # noqa W505
+            Input list of electrometer/photodiode measurement handles
+            to combine.
+
+        Returns
+        -------
+        results : `lsst.pipe.base.Struct`
+            The results struct containing:
+
+            ``outputData``
+                Final combined filter scan, with a single table
+                containing the measured throughput for all input
+                filters at the various wavelength values indicated in
+                the exposure's observationReason
+                (`astropy.table.Table`).
+        """
+        filterScanResults = {}
+        filterSet = set()
+        efdClient = CpEfdClient(efdInstance=self.config.efdClientInstance)
+        monochromatorData = efdClient.getEfdMonochromatorData()
+
+        for handle in inputExpHandles:
+            visitInfo = handle.get(component="visitInfo")
+            metadata = handle.get(component="metadata")
+
+            physical_filter = handle.dataId['physical_filter']
+            exposureId = handle.dataId['exposure']
+            exposureTime = visitInfo.exposureTime
+
+            values = [v for k, v in metadata.items() if "LSST ISR FINAL MEAN" in k]
```

Comment:
I don't love using the `MEAN` here over the full frame.  Either the `median` or (like the PTC) we need to compute the mean with excluding the edges.  (Though it may be that because we're computing a ratio with and without the filter it cancels, I still don't like using the MEAN).

Reply:
Switching to MEDIAN (which should also already be in the headers).